packages/backend/src/open_payments/payment_pointer/model.ts

```diff
@@ -91,3 +91,97 @@ export class PaymentPointerEvent extends WebhookEvent {
   public type!: PaymentPointerEventType
   public data!: PaymentPointerData
 }
+
+export interface GetOptions {
+  id: string
+  clientId?: string
+  paymentPointerId?: string
+}
+
+export interface ListOptions {
+  paymentPointerId: string
+  clientId?: string
+  pagination?: Pagination
+}
+
+class SubresourceQueryBuilder<
+  M extends Model,
+  R = M[]
+> extends BaseModel.QueryBuilder<M, R> {
+  ArrayQueryBuilderType!: SubresourceQueryBuilder<M, M[]>
+  SingleQueryBuilderType!: SubresourceQueryBuilder<M, M>
+  MaybeSingleQueryBuilderType!: SubresourceQueryBuilder<M, M | undefined>
+  NumberQueryBuilderType!: SubresourceQueryBuilder<M, number>
+  PageQueryBuilderType!: SubresourceQueryBuilder<M, Page<M>>
+
+  private byClientId(clientId: string) {
+    // SubresourceQueryBuilder seemingly cannot access
+    // PaymentPointerSubresource relationMappings, so
+    // this.withGraphJoined('grantRef') results in:
+    // missing FROM-clause entry for table "grantRef"
+    return this.join(
+      'grantReferences as grantRef',
+      `${this.modelClass().tableName}.grantId`,
+      'grantRef.id'
+    ).where('grantRef.clientId', clientId)
+  }
+
+  get({ id, paymentPointerId, clientId }: GetOptions) {
+    if (paymentPointerId) {
+      this.where(
+        `${this.modelClass().tableName}.paymentPointerId`,
+        paymentPointerId
+      )
+    }
+    if (clientId) {
+      this.byClientId(clientId)
+    }
```

Comment:
Is this doing two queries if `paymentPointerId` and `clientId` are given?

Reply:
No, this is all a single query builder that isn't executed until we `await`
https://vincit.github.io/objection.js/api/query-builder/#class-querybuilder